mathesar/imports/csv.py

```diff
@@ -35,19 +35,31 @@ def legacy_create_table_from_csv(name, schema, database_key, csv_file):
     return table
 
 
+def get_sv_delimiter(filename):
+    if filename.endswith('.tsv'):
+        return '\t'
+    elif filename.endswith('.csv'):
+        return ','
+    else:
+        # Should never be hit because we only allow csv and tsv extensions
+        raise NotImplementedError("File extension not supported!")
```

Comment:
This will be pretty brittle with real-world files of any sort.  In particular, the extension isn't standardized at all; many CSV-like files are created via excel export, for example, and those get a `.txt` extension by default when they're not comma-separated (comma-separated gets `.csv` as expected).  LibreOffice always uses `.csv` by default for any text-file table with some delimiter.  Google Sheets has `.csv` and `.tsv`, but can't handle any other delimiters.  Etc.

I think we should either infer the file type by actually looking for reasonable delimiters, or take it as input to the API in the request.